src/cookboot/adapters/persistance/recipe_repo_in_memory.py

```diff
@@ -0,0 +1,29 @@
+from cookboot.domain.recipe import Ingridient, Recipe, RecipeId
+from cookboot.domain.user import UserId
+from uuid import uuid4
+
+
+class RecipeRepoInMemory:
+    """In memory implementation of RecipeRepo protocol.
+    Usefull for testing."""
+
+    def __init__(self) -> None:
+        self._repo: dict[RecipeId, Recipe] = dict()
+
+    def create(self, name: str, ingridients: list[Ingridient], user_id: UserId) -> Recipe:
+        id: RecipeId = str(uuid4())
+        recipe = Recipe(id, name, ingridients, user_id)
+        self._repo[id] = recipe
```

Comment:
```suggestion
        self._repo[recipe.id] = recipe
```

Reply:
Rejected - just 2 lines above id is defined with the type RecipeId which is clear enough.